starter-core/src/main/java/io/micronaut/starter/feature/agorapulse/console/Console.java

```diff
@@ -0,0 +1,162 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.agorapulse.console;
+
+import com.fizzed.rocker.RockerModel;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.dependencies.Coordinate;
+import io.micronaut.starter.build.dependencies.Dependency;
+import io.micronaut.starter.feature.Category;
+import io.micronaut.starter.feature.agorapulse.AgoraPulseFeature;
+import io.micronaut.starter.feature.agorapulse.console.template.consoleGroovyDsl;
+import io.micronaut.starter.feature.agorapulse.console.template.consoleGroovyHttp;
+import io.micronaut.starter.feature.agorapulse.console.template.consoleKotlinHttp;
+import io.micronaut.starter.options.BuildTool;
+import io.micronaut.starter.options.Language;
+import io.micronaut.starter.template.RockerTemplate;
+import io.micronaut.starter.util.VersionInfo;
+import jakarta.inject.Singleton;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+@Singleton
+public class Console implements AgoraPulseFeature {
+
+    private static final String ARTIFACT_ID = "micronaut-console";
+    private static final String SSRF_HEADER_NAME = "X-Console-Verify";
+
+    @Override
+    @NonNull
+    public String getName() {
+        return "micronaut-console";
+    }
+
+    @Override
+    public String getTitle() {
+        return "Micronaut Console";
+    }
+
+    @Override
+    public String getThirdPartyDocumentation() {
+        return "https://agorapulse.github.io/micronaut-console/";
+    }
+
+    @Override
+    public boolean supports(ApplicationType applicationType) {
+        return applicationType == ApplicationType.DEFAULT;
+    }
+
+    @Override
+    public String getCategory() {
+        return Category.DEV_TOOLS;
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        String secret = UUID.randomUUID().toString();
+        addDependency(generatorContext);
+        addExampleCode(generatorContext, secret);
+        addConfiguration(generatorContext, secret);
+    }
+
+    private void addDependency(GeneratorContext generatorContext) {
+        generatorContext.addDependency(Dependency.builder()
+                .lookupArtifactId(ARTIFACT_ID)
+                .runtime());
```

Comment:
for Gradle, please use `developmentOnly` not `runtimeOnly`. That way, it is not in production applications by default.

Reply:
sounds good but how exactly can I do it?

Reply:
It should be possible after this PR gets merged. https://github.com/micronaut-projects/micronaut-starter/pull/1215

Reply:
resolved in https://github.com/micronaut-projects/micronaut-starter/pull/1207/commits/e7339697350e512458318de87a355088ad729b42